src/tests/grid/test_grids.cc

```diff
@@ -428,107 +428,192 @@ CASE( "test_structured_from_config" ) {
     EXPECT( not g.domain().global() );
 }
 
+struct PointLatLon: public Point2 {
+    using Point2::Point2;
+    PointLatLon() : Point2() {}
+
+    // Allow initialization through PointXY lonlat = {0,0};
+    PointLatLon( std::initializer_list<double> list ) : PointLatLon( list.begin() ) {}
+
+    double lon() const { return x_[1]; }
+    double lat() const { return x_[0]; }
+};
+
 CASE( "test_cubedsphere" ) {
-    int resolution( 2 );
-    std::vector<std::string> grid_names{
-        "CS-EA-" + std::to_string( resolution ),
-        "CS-ED-" + std::to_string( resolution ),
-        "CS-LFR-" + std::to_string( resolution ),
-    };
-
-    for ( std::string& s : grid_names ) {
-        Grid grid{s};
-        EXPECT( grid );
-        std::vector<PointLonLat> pointLonLats_from_XY;
-        std::vector<PointXY> pointXYs;
-        std::vector<PointLonLat> pointLonLats;
-        std::vector<PointLonLat> pointXYs_from_LonLat;
-
-        for ( auto crd : grid.xy() ) {
-            pointXYs.push_back( crd );
-            grid->projection().xy2lonlat( crd );
-            pointLonLats_from_XY.emplace_back( crd );
-        }
-        for ( auto crd : grid.lonlat() ) {
-            pointLonLats.emplace_back( crd );
-            grid->projection().lonlat2xy( crd );
-            pointXYs_from_LonLat.emplace_back( crd );
-        }
-        EXPECT( pointLonLats.size() == 6 * resolution * resolution + 2 );
-        EXPECT( pointXYs.size() == 6 * resolution * resolution + 2 );
-        EXPECT( grid.size() == 6 * resolution * resolution + 2 );
-
-        // Note that with nodal points on the cubed-sphere
-        // for a equiangular and equidistant projections and a resolution of 2 are the same.
-        if ( resolution == 2 ) {
-            constexpr double rpi     = M_PI;
-            constexpr double rad2deg = 180. / rpi;
-            double cornerLat         = rad2deg * std::atan( std::sin( rpi / 4.0 ) );
-            double tolerance         = 1e-13;
-            // Expected latitudes/longitude per tile
+    constexpr double rad2deg = 180. / M_PI;
+    const double cornerLat = rad2deg * std::atan( std::sin( M_PI / 4.0 ) );
+
+    const std::vector<int> resolutions{1, 2, 4, 8};
+    for( auto resolution : resolutions ) {
+        const std::vector<std::string> grid_names{
+            "CS-EA-L-" + std::to_string( resolution ),
+            "CS-ED-L-" + std::to_string( resolution ),
+            "CS-LFR-L-" + std::to_string( resolution ),
+            "CS-EA-C-" + std::to_string( resolution ),
+            "CS-ED-C-" + std::to_string( resolution ),
+            "CS-LFR-C-" + std::to_string( resolution )
+        };
+
+        for ( const std::string& grid_name : grid_names ) {
+            Log::info() << "Testing " << grid_name << std::endl;
+            if( grid_name == "CS-LFR-L-1" ) {
+                Log::error() << eckit::Colour::red << "TODO: Fix me!!!. Skipping..." << eckit::Colour::reset << std::endl;
+                continue;
+            }
```

Comment:
I added more grids and resolutions here, and detected that "CS-LFR-L-1" gives segmentation faults, hence it is skipped, but should be fixed soon. It must be related to previous comment.